From 80d1f240f9e366f9e8ddfbade3297e55ec8243fc Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 3 Jul 2024 12:12:38 +0200
Subject: [PATCH 0992/1695] plat-stm2mp1: remove redundant pin secure
 configuration in shared resources

The secure state of the pins is already set in the stm32_gpio driver.
Remove this redundant code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4109d0cd2d2ed48773aabc91dd85e62466afe0c5 plat-stm2mp1: remove redundant pin secure configuration in shared resources]
Change-Id: Icac976c5dd62966ba180156426c4884e81b2c377
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391139
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../arch/arm/plat-stm32mp1/shared_resources.c | 28 -------------------
 1 file changed, 28 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/shared_resources.c b/core/arch/arm/plat-stm32mp1/shared_resources.c
index 94640b666..b1d4951bf 100644
--- a/core/arch/arm/plat-stm32mp1/shared_resources.c
+++ b/core/arch/arm/plat-stm32mp1/shared_resources.c
@@ -643,28 +643,6 @@ static void check_rcc_secure_configuration(void)
 	}
 }
 
-static void set_gpio_secure_configuration(void)
-{
-	unsigned int pin = 0;
-
-	for (pin = 0; pin < get_gpioz_nbpin(); pin++) {
-		enum stm32mp_shres shres = STM32MP1_SHRES_GPIOZ(pin);
-		bool secure = stm32mp_periph_is_secure(shres);
-
-		stm32_gpio_set_secure_cfg(GPIO_BANK_Z, pin, secure);
-	}
-}
-
-static TEE_Result gpioz_pm(enum pm_op op, uint32_t pm_hint __unused,
-			   const struct pm_callback_handle *hdl __unused)
-{
-	if (op == PM_OP_RESUME)
-		set_gpio_secure_configuration();
-
-	return TEE_SUCCESS;
-}
-DECLARE_KEEP_PAGER(gpioz_pm);
-
 static TEE_Result stm32mp1_init_final_shres(void)
 {
 	enum stm32mp_shres id = STM32MP1_SHRES_COUNT;
@@ -678,12 +656,6 @@ static TEE_Result stm32mp1_init_final_shres(void)
 		     shres2str_id(id), id, shres2str_state(*state));
 	}
 
-	if (IS_ENABLED(CFG_STM32_GPIO)) {
-		set_gpio_secure_configuration();
-		register_pm_driver_cb(gpioz_pm, NULL,
-				      "stm32mp1-shared-resources");
-	}
-
 	check_rcc_secure_configuration();
 
 	return TEE_SUCCESS;
-- 
2.47.2

